jwtclient/client.go

```diff
@@ -0,0 +1,22 @@
+package jwtclient
+
+import (
+	"errors"
+	"net/http"
+)
+
+// NewJWTClient is http.Client decorator which returns a client with `JWTTransport` transport.
+// `JWTTransport` will wrap the request, generate JWT token and updated Authorization header.
+func NewJWTClient(client *http.Client, opts ...Option) (*http.Client, error) {
```

Comment:
... and I reading further I see you've already got the equivalent of this. So I vote to eliminate this `NewJWTClient` all together - it's not useful enough on its own.